src/components/Recipes/Recipes.jsx

```diff
@@ -1,16 +1,74 @@
 import * as React from "react"
-import { Link } from "react-router-dom"
+import { useState, useEffect } from "react"
+import { Alert } from "react-bootstrap" 
+import axios from 'axios';
 import "./Recipes.css"
 
+import { useAuth } from "../../contexts/AuthContext"
 import RecipeCard from "../RecipeCard/RecipeCard"
 
 export default function Recipes(props) {
+  // get user data from the database
+  const { currentUser } = useAuth()
+  
+  const listUserRecipeUrl = "http://localhost:3001/apiuserrecipes/"
+  const listStdRecipeUrl = "http://localhost:3001/apistdrecipes"
+  const [userInfo, setUserInfo] = useState({})
+  const [recipes, setRecipes] = useState([])
+  const [error, setError] = useState("")
+  
+  var userProducts = []
+  if (userInfo == undefined) {
+    setError("Please log in first")
+  }
+  // TODO: look into why userInfo keeps becoming undefined when refresh (current user is not undefined)
+  else if (userInfo !== undefined && userInfo.data != undefined && userInfo.data.products !== undefined) {
+    var userProducts = userInfo.data.products
+  }
+
+  // get best recipe matches with user's food items
+  useEffect(() => {
+    setUserInfo(props.users.find(u => u.uid === currentUser.uid))
+    async function fetchStdRecipes() {
+      try {
+        var { data } = await axios(listStdRecipeUrl)
+        setRecipes(data)
+      } catch (err) {
+        setError(err.message)
+      }
+    }
+
+    if (userProducts.length === 0) {
+      fetchStdRecipes()
+    }
+  }, [])
```

Comment:
Your undefined issue _might_ be because of this useEffect. Maybe try passing in [currentUser.uid] as the second argument. 
According to this docs, https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects, passing an empty array will only call the useEffect on mount and unmount so maybe when you rerender you're not actually calling this effect and it's defaulting to the initial {} for userInfo.
<img width="753" alt="image" src="https://user-images.githubusercontent.com/57919923/179271318-8553e65c-4f4f-49ac-8e60-244bf8f1b4c5.png">